### PR TITLE
Add jdbc sink

### PR DESCRIPTION
**Pull request by Minutis**
Sink for sinking data to SQL table.

### CODE REVIEW

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java [hunk 1]
```diff
@@ -14,6 +15,10 @@ private Record(Map<String, Object> cells) {
         return (T) cells.get(name);
     }
 
+    public Set<Map.Entry<String, Object>> getCells() {
```
Comment (pdambrauskas) [inline racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: Can we somehow hide internal Record implemetation? We should'nt expose that we are using Map for cells.
Maybe wrap it wit Pair or some custom Pojo?

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java [hunk 2]
```diff
@@ -38,4 +43,11 @@ public Record build() {
             return new Record(cells);
         }
     }
+
+    @Override
+    public String toString() {
```
Comment (pdambrauskas) [inline racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: What do you think about using Pojomatic?

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import org.zapodot.junit.db.EmbeddedDatabaseRule
+import run.racoon.commons.domain.Record
+import run.racoon.tasks.jdbc.source.JdbcSourceTask
+import run.racoon.tasks.jdbc.source.JdbcSourceTaskState
+import spock.lang.Specification
+
+class JdbcSinkTaskTest extends Specification {
+
+    @Rule
+    EmbeddedDatabaseRule dbRule = EmbeddedDatabaseRule
```
Comment (pdambrauskas) [inline racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy]: is it the same variable, we are using on source test? maybe we should create some factory method for it?

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                );
+            """)
+            .build()
+
+    def "Should sink two messages"() {
+        given:
+        def state = new JdbcSinkTaskState(
+                dbRule.getConnectionJdbcUrl(),
+                "INSERT INTO coons (first_name, last_name, age) VALUES (?, ?, ?);",
+                Map.of(
```
Comment (pdambrauskas) [inline racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy]: It's groovy, use `[first_name: 1, last_name: 2, age: 3]`

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                    assert resultSet.getString(2) == 'Coon2'
+                    assert resultSet.getInt(3) == 22
+
+                    assert !resultSet.next()
+                }
+            }
+        }
+    }
+
+    def byName(record, name){
```
Comment (pdambrauskas) [inline racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy]: Not used

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                    .cell("last_name", "Coon2")
+                    .cell("age", 22)
+                    .build()]
+        task.push(records)
+
+        then:
+        dbRule.getConnection().withCloseable { connection ->
+            connection.createStatement().withCloseable { statement ->
+                statement.executeQuery("SELECT * FROM coons").withCloseable { resultSet ->
+                    assert resultSet.next()
```
Comment (pdambrauskas) [inline racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy]: groovy allows you to do something like:
```groovy
resultSet.with {
  assert next()
  assert getString(1) == 'Racoon'
  assert getString(2) == 'Coon'
  ....
}
```
not sure if it makes this block nicer though :)

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java (hunk 1 above)
Comment (Minutis) [reply racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: This operation will be used very often, wrap it somehow might affect performance. For each record, you would need to convert it to some other type. You sure this is a good idea?

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy (hunk 6 above)
Comment (Minutis) [reply racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/sink/JdbcSinkTaskTest.groovy]: ty for the tip, don't think that it is nicer like you said, so I'll just leave it like this.

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java (hunk 1 above)
Comment (pdambrauskas) [reply racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: Yes. Do not be paranoic :). short-lived objects have no noticeable penalty on GC.

File: racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/JdbcTestUtil.groovy [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package run.racoon.tasks.jdbc
+
+import org.zapodot.junit.db.CompatibilityMode
+import org.zapodot.junit.db.EmbeddedDatabaseRule
+
+class JdbcTestUtil {
+    def static createEmbeddedDatabaseRule(String initSql) {
```
Comment (pdambrauskas) [inline racoon-tasks/src/test/groovy/run/racoon/tasks/jdbc/JdbcTestUtil.groovy]: You don't need `def` if you have `static`

File: racoon-commons/src/main/java/run/racoon/commons/domain/Record.java [hunk 8]
```diff
@@ -38,4 +45,9 @@ public Record build() {
...
```
Comment (pdambrauskas) [inline racoon-commons/src/main/java/run/racoon/commons/domain/Record.java]: You forgot `@Autoproperty` annotation for this class

**Pull request merged**